{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "48340363-1cc3-4cfb-aa10-55ceb82ba63d",
   "metadata": {
    "tags": []
   },
   "source": [
    "# 1. precocessiong of mutation data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24c6279f-27b5-4cbd-a2f1-403278950c0f",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#set path\n",
    "setwd('/data/shanlan/crc_trio/1.sub_analysis')\n",
    "out_dir='/data/shanlan/crc_trio/1.sub_analysis/plot/'\n",
    "#create dir\n",
    "dir.create(\"./runtime\")\n",
    "dir.create(\"./runtime/2.sgWGS\")\n",
    "dir.create(\"./runtime/2.sgWGS/tomaf\")\n",
    "dir.create(\"./runtime/2.sgWGS/maf\")\n",
    "dir.create(\"./runtime/2.sgWGS/ccf\")\n",
    "dir.create(\"./runtime/2.sgWGS/output\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7458558c-a0a4-4126-8628-87b08c9315f3",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#set path\n",
    "setwd('/data/shanlan/crc_trio/1.sub_analysis')\n",
    "out_dir='/data/shanlan/crc_trio/1.sub_analysis/plot/'\n",
    "\n",
    "#load library\n",
    "#basic\n",
    "library(openxlsx)\n",
    "library(reshape2)\n",
    "library(dplyr)\n",
    "library(stringr)\n",
    "library(car)\n",
    "\n",
    "#visualization\n",
    "library(ggpubr)\n",
    "library(ggrepel)\n",
    "library(ggpmisc)\n",
    "library(ggsci)\n",
    "library(cowplot)\n",
    "\n",
    "#speciafic\n",
    "library(ComplexHeatmap)\n",
    "library(clusterProfiler)\n",
    "library(maftools)\n",
    "library(MesKit)\n",
    "library(org.Hs.eg.db)\n",
    "library(BSgenome.Hsapiens.UCSC.hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fc4c367-6686-41cf-a30c-b7200e9f591e",
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_dir=\"/data/shanlan/crc_trio/wgsCrypt/results/2.VariantFilter/output\"\n",
    "#1.0 data load\n",
    "filelist<-list.files(path=vcf_dir, pattern = \"somatic.snv.filter.PASS.annovar.all.txt\", full.names = TRUE)\n",
    "#filelist=filelist[grepl('B139',filelist)]\n",
    "length(filelist)\n",
    "datalist <- lapply(filelist, function(x)read.csv(x, sep=\"\\t\",header=T))\n",
    "#samples = read.table(\"./wgsCrypt_sample_pairs.txt\", sep=\"\\t\") #该文件与sample_pairs.txt内容一样\n",
    "\n",
    "name=gsub(paste0(vcf_dir,\"/|_somatic.snv.filter.PASS.annovar.all.txt\"),\"\",filelist)\n",
    "names(datalist)=name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46ad44fa-0b01-4c46-be04-30ecac6d593f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Colon_driver_genes.txt: drivergene_list(n=346) from Prof.Hu. \n",
    "#\"where the putative driver genes were curated by merging all TCGA pan-cancer drivers [refs] (n = 299) and additional CRC-specific drivers (n = 47) annotated by the COSMIC Cancer Gene Census (https://cancer.sanger.ac.uk/census).\"\n",
    "#ref: Bailey, et al. Comprehensive Characterization of Cancer Driver Genes and Mutations. 和COSMIC\n",
    "dr1 <- read.table(\"/data/shanlan/crc_trio/1.sub_analysis/refdb/genelist/Colon_driver_genes.txt\",sep = \"\\t\",header = T)$Gene #n=346, \n",
    "dr2 <- read.table(\"/data/shanlan/crc_trio/1.sub_analysis/refdb/genelist/zq_nc_crcdriver.txt\",sep = \"\\t\",header = F)$V1 #n=46, NC2021, Zhao qi\n",
    "dr3 <- read.table(\"/data/shanlan/crc_trio/1.sub_analysis/refdb/Colon_precancer_genelist_STTT.txt\",sep = '\\t',header = T)$Gene ## n=21 , STTT2024, Shaosen Zhang, et al.\n",
    "top20_crc_wes <- c('APC','TP53','KRAS','FBXW7','LRP1B','PIK3CA','FAT3','FAT4','PCDH17','SMAD4','SOX9','ZFP36L2','ARID1A','TCF7L2','SPTA1','ATM','BRAF','FAT2','MACF1','NBEA','ACVR2A','BCL9L') # n=21, T_wes>5%\n",
    "driverGene <- Reduce(union,list(dr1,dr2,dr3,top20_crc_wes)) \n",
    "driverGene %>% length() #367\n",
    "write.table(data.frame(Gene=driverGene,Tumor='Colon'),file = \"/data/shanlan/crc_trio/1.sub_analysis/refdb/Colon_driver_final.txt\",sep = '\\t',quote = F, col.names = F, row.names = F )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4d6fb54-d14a-4fac-aa9d-f77090f2f191",
   "metadata": {},
   "outputs": [],
   "source": [
    "##rescue APC mutations across all crypts !!!!! # 进一步改进，对所有driver都recall\n",
    "#apc_mut<-read.table('/data/shanlan/crc_trio/wgsCrypt/results/2.Mutect/Solo/sgWGS.Mutect2.APC.txt',header=T,sep='\\t',fill=T)\n",
    "#colnames(apc_mut)=gsub('FAP03_N1_B1_|FAP03_P1_R3_G4_','',colnames(apc_mut))\n",
    "\n",
    "dr1<-read.table('/data/shanlan/crc_trio/wgsCrypt/results/2.Mutect/Solo/sgWGS.Mutect2.Driver.txt',header=T,sep='\\t',fill=T) %>% mutate(mut_id = paste0(Sample, \":\", Chr, \":\", Start, \":\", End, \":\", Ref, \":\", Alt))\n",
    "dr2<-read.table('/data/shanlan/crc_trio/wgsCrypt/results/2.Strelka/sgWGS.Strealk2.Driver.txt',header=T,sep='\\t',fill=T) %>% mutate(mut_id = paste0(Sample, \":\", Chr, \":\", Start, \":\", End, \":\", Ref, \":\", Alt))\n",
    "apc_all <- rbind(dr1, dr2 %>% filter(!mut_id %in% dr1$mut_id) ) %>% dplyr::select(-mut_id) %>% filter(Sample %in% names(datalist) )\n",
    "colnames(apc_all)[7:12]=c(\"N_DP\",\"N_VAF\",\"Total_allele_depth\",\"Ref_allele_depth\",\"Alt_allele_depth\",\"VAF\")\n",
    "\n",
    "#----------------------------------------------------------------------------------------------------\n",
    "# remove false postive mutations\n",
    "# criteria: (1)AD>=3 & DP>=5; or AD>=2 & VAF>=0.1\n",
    "# (2) The site distance to adjacent mutations is greater than 50 bp.  \n",
    "# Specially, 如果两个突变的VAF差距大，保留；另外，对连续插入或缺失需要手动检查\n",
    "#----------------------------------------------------------------------------------------------------\n",
    "tmp1 = apc_all %>% mutate(mut_id = paste0(Chr, \":\", Start, \":\", End, \":\", Ref, \":\", Alt)) \n",
    "tmp1=tmp1 %>% filter((Alt_allele_depth>2 & Total_allele_depth>=5) | (Alt_allele_depth>=2 & VAF>=0.1)) \n",
    "\n",
    "threshold = 50 \n",
    "tmp2= tmp1 %>%  \n",
    "              arrange(Sample,Chr,Start) %>% group_by(Sample,Chr) %>%  \n",
    "              mutate(  \n",
    "                distance_to_previous = Start - lag(Start),  \n",
    "                distance_to_next = lead(Start) - Start,\n",
    "                diffVAF_to_previous = as.numeric(VAF) - lag(as.numeric(VAF)),  \n",
    "                diffVAF_to_next = lead(as.numeric(VAF)) - as.numeric(VAF),\n",
    "                group = cumsum(is.na(distance_to_previous) | distance_to_previous > threshold),\n",
    "                id = paste0(\"id\", group), \n",
    "                LenRef = nchar(Ref),\n",
    "                LenAlt = nchar(Alt) \n",
    "              ) %>%  \n",
    "              mutate(  \n",
    "                to_delete1 = ifelse(  \n",
    "                  (!is.na(distance_to_previous) & distance_to_previous <= threshold) |  \n",
    "                  (!is.na(distance_to_next) & distance_to_next <= threshold), 1, 0 ), \n",
    "                 to_delete2 = ifelse(  \n",
    "                  (!is.na(distance_to_previous) & distance_to_previous <= 10) |  \n",
    "                  (!is.na(distance_to_next) & distance_to_next <= 10), 1, 0), \n",
    "                 to_delete3 = ifelse(  \n",
    "                  (!is.na(diffVAF_to_previous) & abs(diffVAF_to_previous) <= 0.15) |  \n",
    "                  (!is.na(diffVAF_to_next) & abs(diffVAF_to_next) <= 0.15), 1, 0) \n",
    "                 \n",
    "              ) \n",
    "#输出待手动检查突变\n",
    "write.table(tmp2 %>% filter(to_delete1==1), \"./runtime/2.sgWGS/apc_mut_near.txt\", sep = \"\\t\", quote=FALSE, row.names = FALSE)\n",
    "\n",
    "#选择高可信突变\n",
    "apc_hcv<- tmp2 %>% filter(to_delete1!=1)%>% ungroup() %>% dplyr::select(c(1:29)) #dplyr::select(-distance_to_previous, -distance_to_next, -to_delete1)\n",
    "write.table(apc_hcv, \"./runtime/2.sgWGS/apc_mut_hcv.txt\", sep = \"\\t\", quote=FALSE, row.names = FALSE)\n",
    "\n",
    "#重新挽救的邻近突变\n",
    "apc_rec <- read.table('/data/shanlan/crc_trio/1.sub_analysis/runtime/2.sgWGS/apc_mut_near_rescue.txt',header=T,sep='\\t',fill=T)\n",
    "\n",
    "#数据合并\n",
    "apc_mut = rbind(apc_hcv,apc_rec)\n",
    "\n",
    "write.table(apc_mut, \"./runtime/2.sgWGS/rescued_apc_mut_final.txt\", sep = \"\\t\", quote=FALSE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "9b54d420-e176-4876-86e9-ebfdafed898c",
   "metadata": {},
   "outputs": [],
   "source": [
    "###driver突变负荷\n",
    "sn_info <- read.table('/data/shanlan/crc_trio/1.sub_analysis/runtime/2.sgWGS/mt/ST1.mt_sn_info.txt', header = T, sep = '\\t')\n",
    "df = apc_mut %>% left_join(.,sn_info, by = c('Sample'='Tumor_Sample_Barcode'))\n",
    "tt = df %>% dplyr::select(Sample,Derived,Gene.refGene) %>% unique() %>% group_by(Gene.refGene,Derived) %>% tally() %>% arrange(desc(n))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88c3aa63-026f-4854-9125-0bc23efebca8",
   "metadata": {},
   "outputs": [],
   "source": [
    "##1.1 Filtering false postive mutations, please rescue oncogenic APC mutations \n",
    "#运行前把annodir下所有文件都删除了\n",
    "mafdir = '/data/shanlan/crc_trio/1.sub_analysis/runtime/2.sgWGS/tomaf/'\n",
    "unlink(file.path(mafdir, \"*\"), recursive = FALSE)\n",
    "\n",
    "dat=list()\n",
    "for(i in 1:length(datalist)){\n",
    "    tmp1=datalist[[i]] %>% mutate(mut_id = paste0(Chr, \":\", Start, \":\", End, \":\", Ref, \":\", Alt))\n",
    "    colnames(tmp1)[6:11]=c(\"N_DP\",\"N_VAF\",\"Total_allele_depth\",\"Ref_allele_depth\",\"Alt_allele_depth\",\"VAF\")\n",
    "    print(paste0(names(datalist)[i],\":na:\",nrow(tmp1 %>% filter(is.na(N_DP) ))) )\n",
    "    #tmp1=tmp1%>% filter(N_DP >= 10 & Total_allele_depth >= 7 & Alt_allele_depth >= 4 & VAF >= 0.15)\n",
    "    tmp1=tmp1%>% filter(VAF >= 0.15) #because of refiltered-samples with low depth\n",
    "    \n",
    "    tmp2=apc_mut %>% filter(Sample %in% names(datalist)[i]) %>% dplyr::select(-Sample) %>% mutate(mut_id = paste0(Chr, \":\", Start, \":\", End, \":\", Ref, \":\", Alt)) %>% filter(!mut_id %in% tmp1$mut_id)\n",
    "    colnames(tmp2)=colnames(tmp1)\n",
    "    \n",
    "    tmp=rbind(tmp1,tmp2) %>% distinct(.keep_all = TRUE) %>% mutate(Tumor_Sample_Barcode=names(datalist)[i]) %>% dplyr::select(Tumor_Sample_Barcode, mut_id, everything())  \n",
    " \n",
    "    tmp=tmp %>% \n",
    "     filter(is.na(ExAC_ALL) | ExAC_ALL<0.05 ) %>%\n",
    "     filter(is.na(ExAC_EAS) | ExAC_EAS<0.05 ) %>%\n",
    "     filter(is.na(X1000g2015aug_all) | X1000g2015aug_all<0.05 ) %>% \n",
    "     filter(is.na(esp6500siv2_all) | esp6500siv2_all<0.05 ) \n",
    "    \n",
    "    dat[[i]]=tmp\n",
    "    ##duplicate mutation info for each sample\n",
    "    dup_mut=tmp %>% group_by(mut_id) %>% tally() %>% filter(n>1)\n",
    "    print(paste0(names(datalist)[i],\":dup:\",nrow(dup_mut)))\n",
    "}\n",
    "names(dat)=names(datalist)\n",
    "lapply(dat, function(x)write.table(x,paste(\"./runtime/2.sgWGS/tomaf/\", x$Tumor_Sample_Barcode[1], \".txt\",sep=\"\"),sep=\"\\t\",quote=FALSE, row.names = FALSE))\n",
    "\n",
    "##重新读入处理好的突变文件， for maftools，找高频的突变\n",
    "toMaflist<-list.files(path=\"./runtime/2.sgWGS/tomaf\", pattern = \".txt\", full.names = TRUE)\n",
    "#toMaflist<-toMaflist[grepl('B139',toMaflist)]\n",
    "Maflist <- lapply(toMaflist, function(x)annovarToMaf(x,Center = 'NA', refBuild = 'hg38', tsbCol = 'Tumor_Sample_Barcode',table = 'refGene',sep = \"\\t\"))\n",
    "names(Maflist)=gsub(\"./runtime/2.sgWGS/tomaf/|.txt\",\"\",toMaflist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71d6c998-58a7-40d1-a947-1eb39b002810",
   "metadata": {},
   "outputs": [],
   "source": [
    "##重新读入处理好的突变文件， for maftools，找高频的突变\n",
    "toMaflist<-list.files(path=\"./runtime/2.sgWGS/tomaf\", pattern = \".txt\", full.names = TRUE)\n",
    "#toMaflist<-toMaflist[grepl('B139',toMaflist)]\n",
    "Maflist <- lapply(toMaflist, function(x)annovarToMaf(x,Center = 'NA', refBuild = 'hg38', tsbCol = 'Tumor_Sample_Barcode',table = 'refGene',sep = \"\\t\"))\n",
    "names(Maflist)=gsub(\"./runtime/2.sgWGS/tomaf/|.txt\",\"\",toMaflist)\n",
    "names(Maflist)\n",
    "names(Maflist) %>% length()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53d5b19d-6860-480c-9e2e-5fe885d8cc4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical<-data.frame(Tumor_Sample_Barcode= names(Maflist) ) %>% mutate(\n",
    "    Patient = ifelse(grepl(\"^B\",Tumor_Sample_Barcode),stringr::str_sub(Tumor_Sample_Barcode,1,4), \n",
    "                     stringr::str_sub(Tumor_Sample_Barcode,1,5) ),\n",
    "    Tissue = ifelse(grepl(\"N\",Tumor_Sample_Barcode),\"N\",\n",
    "                    ifelse(grepl(\"^B\",Tumor_Sample_Barcode),\n",
    "                           stringr::str_sub(Tumor_Sample_Barcode,5,6) %>% gsub(\"_\",\"\",.), \n",
    "                           stringr::str_sub(Tumor_Sample_Barcode,7,8) ) ), \n",
    "    Tumor=paste0(Patient,Tissue),\n",
    "    Tumor_Sample_Label = ifelse(grepl(\"^B\",Tumor_Sample_Barcode),stringr::str_sub(Tumor_Sample_Barcode,5,-1), stringr::str_sub(Tumor_Sample_Barcode,7,-1) ),\n",
    "    Label=ifelse(grepl(\"^B\",Tumor_Sample_Barcode),stringr::str_sub(Tumor_Sample_Barcode,6+nchar(Tissue),-1), stringr::str_sub(Tumor_Sample_Barcode,10,-1) ),\n",
    "    Region=stringr::str_sub(Label,1,2),\n",
    "    Derived=substr(Tissue,1,1)\n",
    ") %>% mutate(tmp=ifelse(grepl(\"^B\",Tumor), Tumor, paste0(substr(Tumor,1,5),\"_\",substr(Tumor,6,nchar(Tumor) ))), Tumor=tmp ) %>% dplyr::select(-tmp)\n",
    "write.table(clinical, \"./runtime/2.sgWGS/sgWGS_clinical_all.txt\", sep = \"\\t\", quote=FALSE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf04da91-c7df-419c-ade8-d52896e143ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "maf=Maflist                 \n",
    "maf_flatten<-do.call(\"rbind\", lapply(maf, as.data.frame)) %>%\n",
    "             mutate_at(.vars = vars(N_DP, N_VAF, Total_allele_depth, Ref_allele_depth, Alt_allele_depth, VAF), .fun = as.numeric)\n",
    "#maf_flatten$Chromosome<-as.character(maf_flatten$Chromosome)\n",
    "write.table(maf_flatten, \"./runtime/2.sgWGS/sgWGS_maf_all\", sep = \"\\t\", quote=FALSE, row.names = FALSE) #如果meskit输入有ccf文件，则maf文件不能包含CCF列，不然报错，错误: Problem with `mutate()` input `CCF`.\n",
    "\n",
    "write.table(maf_flatten[which(maf_flatten$Variant_Type==\"SNP\"),], \"./runtime/2.sgWGS/sgWGS_maf_all_SNP\", sep = \"\\t\", quote=FALSE, row.names = FALSE) #如果meskit输入有ccf文件，则maf文件不能包含CCF列，不然报错，错误: Problem with `mutate()` input `CCF`.\n",
    "\n",
    "maf_flatten_exonic = subset(maf_flatten[which(maf_flatten$Variant_Type==\"SNP\"),],Variant_Classification == c(\"Missense_Mutation\",\"Silent\",\"Nonsense_Mutation\"))\n",
    "write.table(maf_flatten_exonic, \"./runtime/2.sgWGS/sgWGS_maf_exonic_SNP\", sep = \"\\t\", quote=FALSE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9eb91820-d597-4356-a1fc-0c96aa53ae16",
   "metadata": {},
   "outputs": [],
   "source": [
    "#--------------------------------------------------------------------------------------\n",
    "# ###oldrevision: re-analysis for FAP01/FAP03, excluded FAP03_N2_R4_G6, filter lesions with sn<4\n",
    "#2024-12-13: maf share with Christina\n",
    "#Save the file format as before\n",
    "#----------------------------------------------------------------------------------------\n",
    "library(dplyr)\n",
    "#black list: 综合多个条件来确定\n",
    "black_list<-read.table(\"/data/shanlan/crc_trio/1.sub_analysis/sgWGS_black_list.txt\", sep = \"\\t\", header = F)$V1\n",
    "black_list %>% sort()\n",
    "black_list %>% length()\n",
    "\n",
    "##exclude blacklist\n",
    "maf=Maflist[!names(Maflist) %in% black_list] \n",
    "#maf=Maflist\n",
    "maf_flatten<-do.call(\"rbind\", lapply(maf, as.data.frame)) %>%\n",
    "             mutate_at(.vars = vars(N_DP, N_VAF, Total_allele_depth, Ref_allele_depth, Alt_allele_depth, VAF), .fun = as.numeric) %>%\n",
    "             dplyr::select(Chromosome, Start_Position, End_Position, Reference_Allele, Tumor_Seq_Allele2, Tumor_Sample_Barcode, everything()) \n",
    "\n",
    "shared_lesions=c('FAP01_N','FAP01_P1','FAP01_P2','FAP01_P4','FAP01_P5','FAP01_P6','FAP01_T2','FAP01_T3','FAP03_N','FAP03_P1','FAP03_P2','FAP03_P4','FAP03_P5')\n",
    "shared_pts=c('FAP01','FAP03')\n",
    "#for Ryan\n",
    "TREEDIR=\"/data/shanlan/crc_trio/1.sub_analysis/runtime/2.sgWGS/tree/\"\n",
    "combn_files <- list.files(path=TREEDIR, pattern = \"_clean_meskit\", full.names = TRUE)\n",
    "combn_files <- combn_files[grep('FAP01_|FAP03_',combn_files)]\n",
    "combn_files\n",
    "\n",
    "f_min= list()\n",
    "\n",
    "for(i in combn_files ) {\n",
    "    #i=1\n",
    "    sn<- read.table(i,head=T,sep='\\t')\n",
    "    sn <- sn %>% filter(Tumor_Sample_Barcode != \"FAP03_N2_R4_G6\") %>% filter(!grepl('FAP01_P3|FAP01_T1|FAP03_P3|FAP03_P6|FAP03_P7|FAP03_P8',Tumor_Sample_Barcode) ) #|_B1\n",
    "    objects<-gsub(paste0(TREEDIR,\"/|_clinical_clean_meskit\"),\"\",i)\n",
    "    mut <- maf_flatten %>% filter(Tumor_Sample_Barcode %in% sn$Tumor_Sample_Barcode)\n",
    "    f_min[[objects]] = mut %>% filter(VAF<0.15)\n",
    "    write.table(mut, paste0(TREEDIR,objects,\".all_maf_v2.txt\"), sep = \"\\t\", quote=FALSE, row.names = FALSE,col.names = TRUE) \n",
    "}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06b20cf9-5cb0-47e6-a5c7-3a4f1905b2a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "##rescued list to CC\n",
    "library(dplyr)\n",
    "#black list: 综合多个条件来确定\n",
    "black_list<-read.table(\"/data/shanlan/crc_trio/1.sub_analysis/sgWGS_black_list.txt\", sep = \"\\t\", header = F)$V1\n",
    "black_list %>% sort()\n",
    "black_list %>% length()\n",
    "apc_mut<-read.table( \"/data/shanlan/crc_trio/1.sub_analysis/runtime/2.sgWGS/rescued_apc_mut_final.txt\", sep = \"\\t\", header = T)\n",
    "apc_mut %>% dim()\n",
    "\n",
    "shared_lesions=c('FAP01_N','FAP01_P1','FAP01_P2','FAP01_P4','FAP01_P5','FAP01_P6','FAP01_T2','FAP01_T3','FAP03_N','FAP03_P1','FAP03_P2','FAP03_P4','FAP03_P5')\n",
    "shared_pts=c('FAP01','FAP03')\n",
    "\n",
    "apc_mut_toCC <- apc_mut %>% filter(!Sample %in% black_list) %>% filter(grepl(paste0(shared_lesions,collapse = \"|\"),Sample))\n",
    "apc_mut_toCC %>% dim()\n",
    "write.table(apc_mut_toCC, \"/data/shanlan/crc_trio/1.sub_analysis/runtime/2.sgWGS/rescued_apc_mut_FAP.txt\", sep = \"\\t\", quote=FALSE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33c2fdfe-1fec-4a53-ab4c-5b5dacbe47e0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R4.3",
   "language": "R",
   "name": "ir_r4.3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
